lib/tether/ops_linux.go

```diff
@@ -960,14 +960,32 @@ func (t *BaseOperations) MountTarget(ctx context.Context, source url.URL, target
 	}
 	rawSource.WriteString(source.Path)
 
-	// NOTE: by default we are supporting "NOATIME" and it can be configurable later. this must be specfied as a flag.
-	// Additionally, we must parse out the "ro" option and supply it as a flag as well for this flavor of the mount call.
-	if err := Sys.Syscall.Mount(rawSource.String(), target, nfsFileSystemType, syscall.MS_NOATIME, mountOptions); err != nil {
-		log.Errorf("mounting %s on %s failed: %s", source.String(), target, err)
-		return err
+	if ip := net.ParseIP(source.Hostname()); ip != nil {
+		if err := syscallMount(rawSource.String(), target, nfsFileSystemType, mountOptions); err != nil {
+			log.Errorf("mounting %s on %s failed: %s", source.String(), target, err)
+			return err
+		}
+		return nil
 	}
 
-	return nil
+	log.Debugf("Looking up host %s", source.Hostname())
+	ips, err := net.LookupIP(source.Hostname())
+	if err != nil {
+		log.Errorf("failing to resolve %s. mounting %s on %s failed: %s", source.Hostname(), source.String(), target, err)
+		return err
+	}
+	for _, ip := range ips {
+		//NOTE: the mountOptions of syscall mount only accept addr=ip. addr=FQDN doesn't work
+		//We resolve the ip address nearest the mounting action.
+		mountOptionsIP := strings.Replace(mountOptions, source.Hostname(), ip.String(), -1)
```

Comment:
Are there edge cases where `mountOptions` could contain the `source.Hostname()` multiple times? Could someone provide a `source.Hostname()` like `addr` (a valid hostname, but not a valid FQDN) that actually appears as a _key_ in addition to being a value)?

Maybe prefixing the old/new with `addr=` would reduce risk here?

`strings.Replace(mountOptions, "addr="+source.Hostname(), "addr="+ip.String(), -1)`

Reply:
Yeah, that's really an awesome advice. 
Btw,  currently the mountOptions is not configurable.
It is always `nfsMountOpitons + ",addr=" + host.Host`, and the `nfsMountOptions` is a hard-coded constant.

Reply:
>  Btw, currently the mountOptions is not configurable.

I see! I think this change is still useful to make to avoid potential issues in the future.